tests/test_GitRepository.cc

```diff
@@ -575,19 +578,110 @@ TEST_CASE("GitRepository add() with glob", "[GitWrapper]")
     }
 }
 
+TEST_CASE("GitRepository: get_remote(), add_remote()", "[GitRepository]")
+{
+    std::filesystem::remove_all(reporoot);
+
+    constexpr auto* repo_url
+        = "https://gitlab.desy.de/jannik.woehnert/taskolib_remote_test.git";
+
+    GitRepository repo{ reporoot };
+
+    auto maybe_remote = repo.get_remote("origin");
+    REQUIRE(maybe_remote.has_value() == false);
+
+    auto remote = repo.add_remote("origin", repo_url);
+    REQUIRE(remote.get() != nullptr);
+
+    maybe_remote = repo.get_remote("origin");
+    REQUIRE(maybe_remote.has_value());
+    REQUIRE(maybe_remote->get_name() == "origin");
+    REQUIRE(maybe_remote->get_url() == repo_url);
+    REQUIRE(maybe_remote->get() != nullptr);
```

Comment:
Maybe do all these tests also on `remote` of line 594?
Maybe superfluous, but who knows if `add_remote` fills `Remote` properly only because `get_remote` does?

Reply:
Yep, done.